runtime/lua/vim/pack.lua

```diff
@@ -328,23 +308,24 @@ local function normalize_plugs(plugs)
   for _, p_data in pairs(plug_map) do
     res[p_data.id] = p_data.plug
   end
+  assert(#res == n)
```

Comment:
Out of curiosity, do you remember why this is needed?

Reply:
I am not opposed to this `assert()`, just couldn't understand why it is needed. I thought that this part, which only adds incrementally without holes, should be convenient enough:

```lua
  local plug_map = {}
  local n = 0
  for _, p in ipairs(plugs) do
    -- Collect
    if not plug_map[p.path] then
      n = n + 1
      plug_map[p.path] = { plug = p, id = n }
    end
-- ...
```